### ISSUE TITLE
ObjC method invocations using dot syntax with no parameter are not converted to Swift function calls

### ISSUE THREAD

**Issue by emish** [Swiftify-Corp/Swiftify#220]
Example: 
```
@interface Foo
- (void)doBar;
@end

// Convert the following
someFoo.doBar;

// Results in the following
someFoo.doBar

// Expected the following
someFoo.doBar()
```

**Comment by emish** [comment]
I realize that I did this conversion without the file that defines `Foo`. It is possible there is no way to know if it is a property access or a parameter-less function call without the source file.

**Comment by alex-swiftify** [comment]
Hi @emish, thanks for your suggestion!

Right you are - for the compiler to "know" that `doBar` is a method rather than a property, you need to include the declaration of `Foo` in the same conversion.
Refer to [this sample](https://swiftify.me/h364k7) that works as you probably expect.

If the declaration of `Foo` is in a separate file, you should include this declaration in the same conversion, by either:
- converting all files together in our [web-based project converter](https://swiftify.com/converter/project/)
- selecting all these files and converting them together in the Advanced Project Converter app
- using the Finder extension

If you don't include the declaration of `Foo`, `someFoo.doBar` is indeed using a property invocation (dot) syntax in Objective-C, thus it's converted to a property invocation in Swift: https://swiftify.me/h364k7/1

Note that a correct way to invoke a method in Objective-C is by using a square brackets syntax: `[someFoo doBar];`.
This is also translated as you expect: https://swiftify.me/h364k7/2

To summarize, your code is using a property access syntax to invoke a method in Objective-C, which is supported, but without the knowledge of the underlying type there's absolutely no way for Swiftify to infer that `doBar` is actually a method name.

I'm marking this ticket as closed, but don't hesitate to write back if you have any additional concerns.
